### PR TITLE
No need for a return statement that says "return witness" when an identical statement is underneath it.

### PR DESCRIPTION
**Pull request by ghost**
The code looks cleaner without having a "return witness" statement in the if bracket while the very next line is "return witness". Having a redundant statement is unnecessary.

### CODE REVIEW

File: lib/AST/ASTContext.cpp [hunk 1]
```diff
@@ -1069,7 +1069,6 @@ ASTContext::getBuiltinInitDecl(NominalTypeDecl *decl,
   if (!witness) {
     assert(false && "Missing required witness");
     witness = ConcreteDeclRef();
```
Comment (CodaFi) [inline lib/AST/ASTContext.cpp]: I think this entire block can be folded to `assert(witness && "Missing required witness")`.

Comment (pi1024e) [reply lib/AST/ASTContext.cpp]: How so?

Comment (AnthonyLatsis) [reply lib/AST/ASTContext.cpp]: The assert is always hit.

**Comment by pi1024e** [comment]
@CodaFi Is this good now?

**Review by AnthonyLatsis — COMMENTED** [review]
I have a hunch these assertions were like that on purpose to only hit on certain build configs,  so leaving the approval to @CodaFi

**Comment by AnthonyLatsis** [comment]
@swift-ci please smoke test

File: lib/AST/ASTContext.cpp [hunk 2]
```diff
@@ -1058,19 +1058,11 @@ ASTContext::getBuiltinInitDecl(NominalTypeDecl *decl,
   auto builtinProtocol = getProtocol(builtinProtocolKind);
   auto builtinConformance = getStdlibModule()->lookupConformance(
       type, builtinProtocol);
-  if (builtinConformance.isInvalid()) {
-    assert(false && "Missing required conformance");
-    witness = ConcreteDeclRef();
-    return witness;
-  }
+  assert(!builtinConformance.isInvalid && "Missing required conformance");
```
Comment (theblixguy) [inline lib/AST/ASTContext.cpp]: I think you forgot the parens:

```suggestion
  assert(!builtinConformance.isInvalid() && "Missing required conformance");
```

**Comment by xwu** [comment]
Agree with @anthonylatsis: This does not look equivalent to the original code. The returned value would change if built without assertions.

**Comment by CodaFi** [comment]
The returned value wouldn't get too far if the first folded block failed its (now former) precondition. I thought this change was limited to the second block...

**Comment by CodaFi** [comment]
Oh no, I see this code is also writing into the conformance on failure. We cannot fold these blocks, we should leave this code alone.

**Comment by pi1024e** [comment]
@theblixguy thank you.
@AnthonyLatsis Could you run "please test"?

**Review by CodaFi — CHANGES_REQUESTED** [review]
I don't think we should touch this code. This changes the observable behavior of this function too deeply in non-asserts configurations, and requires further coordination with the clients of this function.

**Pull request closed without merging**